pandas/tests/sparse/frame/test_apply.py

```diff
@@ -90,3 +91,18 @@ def test_applymap(frame):
     # just test that it works
     result = frame.applymap(lambda x: x * 2)
     assert isinstance(result, SparseDataFrame)
+
+
+def test_apply_toindex():
+    # GH 21539
+    tmp = np.array([[1, 1, 1], [1, 0, 0], [1, 1, 0], [1, 1, 0]])
+    tmp_df = pd.DataFrame(tmp)
+    result = tmp_df.apply(lambda x: x[x == 1].index.tolist(), axis=1)
+    expected = pd.Series([[0, 1, 2], [0], [0, 1], [0, 1]])
```

Comment:
@uds5501 this is still in the wrong file, needs to be in pandas/tests/frame/test_apply.py (its in sparse), and really check whether this actually duplicates current tests.

Reply:
@jreback  Oh! okay! thank you for guidance. Will check it rightaway

Reply:
@jreback @gfyoung 
 I looked into the frame/test_apply.py and found out that most of the tests are correlating to each other. My test is just unique in application of `index.to_list` and asserting equality with Series, shall I still add these tests in those files?

Reply:
as I said, I don't this test actually adds anything, this is just return a list in the lambda.